tests/lib/mem_alloc/testcase.yaml

```diff
@@ -8,3 +8,8 @@ tests:
     arch_exclude: posix
     platform_exclude: qemu_x86_64 # No newlib
     tags: clib newlib
+  libraries.libc.newlibnano:
+    extra_args: CONF_FILE=prj_newlibnano.conf
+    arch_whitelist: arm
```

Comment:
I think you need a toolchain whitelist, I believe I observed that nano was not supported by the zephyr-sdk (but gnuarmemb does support it).